libvast/src/segment_store.cpp

```diff
@@ -352,12 +345,34 @@ void segment_store::inspect_status(caf::settings& dict) {
   put(current, "size", builder_.table_slice_bytes());
 }
 
-segment_store::segment_store(path dir, uint64_t max_segment_size,
-                             size_t in_memory_segments)
-  : dir_{std::move(dir)},
-    max_segment_size_{max_segment_size},
-    cache_{in_memory_segments} {
-  // nop
+caf::error segment_store::register_segments() {
+  for (auto filename : directory{segment_path()})
+    if (auto err = register_segment(filename))
+      return err;
+  return caf::none;
+}
+
+caf::error segment_store::register_segment(const path& filename) {
+  auto chk = chunk::mmap(filename);
+  if (!chk)
+    return make_error(ec::filesystem_error, "failed to mmap chunk", filename);
+  auto s = fbs::GetSegment(chk->data());
```

Comment:
`s` needs to be checked for `nullptr`.

Reply:
Why? The operation cannot fail. It just so happens that flatbuffers always returns a pointer.

Reply:
Maybe you can add `VAST_ASSERT(s)` in the line below for clarity.

Reply:
The thing is that all `Get*` can never be `nullptr`, by design. The function always returns a pointer value.

Reply:
That was not clear to me, a pointer as return type signals the opposite to me as a reader after all.